src/parser.ts

```diff
@@ -3927,6 +3931,14 @@ export function parseMemberOrUpdateExpression(
       case Token.Period: {
         nextToken(parser, (context | Context.AllowEscapedKeyword | Context.InGlobal) ^ Context.InGlobal);
 
+        if (
+          context & Context.InClass &&
+          (parser.token & Token.PrivateField) === Token.PrivateField &&
```

Comment:
I am confused, why not do `parser.token === Token.PrivateField`?

Reply:
Well, based on how `parser.token` is used in other places I made an assumption that `parser.token` might contain multiple values combined together and therefore bitwise AND is required, for example the same method does similar comparison located at the very beginning of it:
```js
if ((parser.token & Token.IsUpdateOp) === Token.IsUpdateOp && (parser.flags & Flags.NewLine) === 0) {
    expr = parseUpdateExpression(parser, context, expr, start, line, column);
  } else if ((parser.token & Token.IsMemberOrCallExpression) === Token.IsMemberOrCallExpression) {
...
```

Reply:
IsUpdateOp is different, it's not a token, but a trait shared by few real token types. Those traits are using higher bits (bit 13 to 31).

```js
  Assign                  = 29 | IsAssignOp | IsClassField, // =
  ShiftLeftAssign         = 30 | IsAssignOp, // <<=
  ShiftRightAssign        = 31 | IsAssignOp, // >>=
  LogicalShiftRightAssign = 32 | IsAssignOp, // >>>=
```

PrivateField doesn't turn on any of those traits, 131 is well below bit 13.
```js
PrivateField      = 131,
```

Reply:
Ah, I see your point! 👍 Thanks, I will correct then!

Reply:
Changed in a056767af5ad0ec8124ca0bc988dc4fd32012fe3